lib/type/string.ex

```diff
@@ -7,6 +7,10 @@ defmodule Antikythera.VersionStr do
   @moduledoc """
   Format of versions of antikythera instances and gears.
 
+  Mostly generated by using `Antikythera.MixCommon.version_with_last_commit_info/1` defined in `mix_common.exs` file.
+  It forces incrementing OTP application versions of antikythera instances or gears according to current time and git commit hash.
+  Useful if you are going to continuously deploy them on every commit to target branches.
```

Comment:
Explaining the rationale of the version string format here is a good idea, but we should make things more accurate.
e.g. we don't use current time (if so it'd always make a new version just by recompiling); we use committer date.
Try this:

```
The format rule is a stricter variant of [semantic versioning](https://semver.org/);
pre-release part and build metadata part are filled with information from the current git commit (committer date and commit hash, respectively).
This way we always make a new version for each new git commit, which enables both antikythera instances and gears to be deployed on a per-commit basis.
The actual version strings are created by `Antikythera.MixCommon.version_with_last_commit_info/1` defined in `mix_common.exs` file.
```

Reply:
Looking good. de8226a
Current time thing was a foolish error on my part..